### PR TITLE
ESLint: no-restricted-syntax

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
Enable no-restricted-syntax rule

### TEST PLAN
Run `npm run lint` and verify there are no new linter errors

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by nytai — APPROVED** [review]

**Comment by nytai** [comment]
@kgabryje is this ready to merge or is it still WIP?

**Comment by kgabryje** [comment]
@nytai It's still WIP

**Comment by kgabryje** [comment]
@nytai It's ready now, I added some more changes. There are more changes required for no-restricted-syntax rule, but I will push them in separate PR to keep this one small

File: superset-frontend/src/explore/controlUtils.js [hunk 1]
```diff
@@ -52,21 +52,18 @@ export function validateControl(control, processedState) {
...
-          typeof control === 'object' &&
-          control.name === controlKey
-        ) {
-          return control;
-        }
-      }
-    }
-  }
-  return null;
+  return (
```
Comment (mistercrunch) [inline superset-frontend/src/explore/controlUtils.js:55]: This one is much more opaque to me and this seems more risky than the other pieces of logic in this PR.  I checked and there is not unit test on this method. Can we either 1) write a covering unit test that works before/after or 2) minimize the changes. 

It looks like there are existing fixtures for `controlPanelSections` that can be refactored out of superset-frontend/spec/javascripts/explore/controlUtils_spec.jsx .

Comment (kgabryje) [reply superset-frontend/src/explore/controlUtils.js:55]: I added some tests and created a separate fixtures file for controlPanelSections

Comment (kgabryje) [reply superset-frontend/src/explore/controlUtils.js:55]: They pass on both old and refactored version of this function

**Review by mistercrunch — APPROVED** [review]
banana

**Comment by ktmud** [comment]
Sorry I noticed that the "Prefer Typescript" hook was not able to detect new `jsx` files so was trying to work on a fix. Rebased the branch and pushed by accident. But the rebase shouldn't contain any new code.

**Pull request merged**